src/workerd/jsg/setup.h

```diff
@@ -683,12 +693,12 @@ class Isolate: public IsolateBase {
 
     // Sets an env value that will be expressed on the process.env
     // if/when nodejs-compat mode is used.
-    void setEnvField(const JsValue& name, const JsValue& value) override {
-      getEnv().set(*this, name, value);
+    void setProcessEnvField(const JsValue& name, const JsValue& value) override {
+      getProcessEnv().set(*this, name, value);
     }
 
     // Returns the env base object.
-    JsObject getEnv(bool release = false) override {
+    JsObject getProcessEnv(bool release = false) override {
```

Comment:
Could we actually change the `process.env` implementation to be based on the new `getWorkerEnv()`, rather than hooking into `workerd-api`? It seems like this should be functionally identical, except if `disallow_importable_env`, but I think that's a feature, not a bug: I would like this flag to prevent my env from being globally available by any means. I'd rather not have to set multiple flags.

I'm OK with that being a separate PR but it would need to be done soon.